gradio/utils.py

```diff
@@ -1466,3 +1459,23 @@ def __len__(self):
 
     def as_list(self):
         return [v for _, v in self.data]
+
+
+def safe_join(directory: DeveloperPath, path: UserProvidedPath) -> str:
```

Comment:
Using these custom types so that in the future user provided paths are not accidentally sanitized before being joined